application/server/configuration.go

```diff
@@ -247,7 +246,10 @@ func updateDeltaFindings(c *config.Config, settings types.Settings) {
 		enable = false
 	}
 
-	c.SetDeltaFindingsEnabled(enable)
+	modified := c.SetDeltaFindingsEnabled(enable)
+	if modified {
+		sendDiagnosticsForAllProducts(c)
```

Comment:
I'm wondering, if we can NOT send diagnostics from configuration. This does not belong to configuration, it would be better to publish the change (modified) event on a channel and have a goroutine outside of configuration act on it. WDYT?